lib/data/models/movie_model.dart

```diff
@@ -0,0 +1,82 @@
+import 'package:json_annotation/json_annotation.dart';
+import 'package:equatable/equatable.dart';
+
+part 'movie_model.g.dart';
+
+@JsonSerializable(createToJson: false)
+class MovieModel extends Equatable{
+  List<Results>? results;
+  int? page;
+  int? totalPages;
+  int? totalResults;
+
+  MovieModel({
+    this.page,
+    this.results,
+    this.totalPages,
+    this.totalResults,
+  });
+
+  factory MovieModel.fromJson(Map<String, dynamic> map) =>
+      _$MovieModelFromJson(map);
+
+  @override
+  List<Object?> get props =>[results];
+}
+
+@JsonSerializable(createToJson: false)
+class Results extends Equatable{
+  @JsonKey(name: "title")
```

Comment:
@Aayushi12345 I suggest don't use JsonKey for all the variables untill it will voilet the naming convention rules.